### DISCUSSION TITLE
Switch to console in macro?

### DISCUSSION THREAD

**Discussion by AJolly** [xblax/flashforge_ad5m_klipper_mod#222] [Ideas]
Is there a way to trigger the UI to switch the screen to display console in a macro? 

For example, id like to have klipper screen switch to show the console if I run this macro.  

[gcode_macro a_SCREWS_TILT_CALCULATE]
gcode:
    G28
    SCREWS_TILT_CALCULATE

**Comment by xblax** [comment]
@AJolly 
I've converted this into a discussion, since it's not really issue. 

I'm not aware of such functionality in klipper screen, to change UI state from klipper directly. That feature would have to be implemented in klipper screen I assume to react on some event.

**Reply by consp** [reply]
Yes, there are already some triggers, like starting a print, which do this. Both KlipperScreen and GuppyScreen have some method for doing this.

It requires a deep code changes and this would be better suited for the KlipperScreen repo since it's a fundamental change and not printer related, same for GuppyScreen. 

Somewhere in August I have some days  off and I'll sync all the repos's again and merge the changes. Shouldn't be too much of an issue but it requires some testing. I'll have a look into making it into a patch for the original repo instead of a fork to make updating easier in the future.
